### PR TITLE
docs(demo): add demo component for collapse

### PR DESCRIPTION
**Pull request by sirajc**
Adding Demo Components for collapse as part of #264

### CODE REVIEW

File: demo/src/app/components/collapse/demos/default-close/default.close.collapse.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+import { NgbCollapse } from '@ng-bootstrap/collapse';
+
+@Component({
+  selector: 'ngbd-collapse-default-close',
+  template: require('./default.close.collapse.html'),
+  directives: [NgbCollapse]
+})
+export class CollapseDefaultCloseComponent {
+  isCollapsed: boolean = true; // Keep content collapsed by default
```
Comment (icfantv) [inline demo/src/app/components/collapse/demos/default-close/default.close.collapse.ts]: should be `private`, don't need data type as it's inferred from the initial value.

File: demo/src/app/components/collapse/demos/basic/basic.collapse.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+import { NgbCollapse } from '@ng-bootstrap/collapse';
+
+@Component({
+  selector: 'ngbd-collapse-basic',
+  template: require('./basic.collapse.html'),
+  directives: [NgbCollapse]
+})
+export class CollapseBasicComponent {
+  isCollapsed: boolean = false;
```
Comment (icfantv) [inline demo/src/app/components/collapse/demos/basic/basic.collapse.ts]: don't need the data type here as it's inferred.

**Comment by icfantv** [comment]
@sirajc, a couple notes for multiple files:
1. please ensure there's a newline at the end of every file
2. please match the existing naming convention for files and directories (e.g. `default.close.collapse.ts` ==> `default.collapse.ts` and `default-close` ==> `default`)

**Comment by icfantv** [comment]
Unless the rest of the team has anything else to add, other than my notes, this LGTM.

**Comment by Foxandxss** [comment]
My only concerns is what we talked already, 20 demos for one component. Do we really need a demo where by default is open and another where by default is closed? Is too much imo.

**Comment by pkozlowski-opensource** [comment]
Agree with @Foxandxss - in case of this simple directive just one demo is enough (we can stick with the default open).

@sirajc could you please make the changes corresponding to the comments so we can merge it? Thnx!

**Comment by sirajc** [comment]
Ok, will update PR as per comments provided by you all

**Comment by sirajc** [comment]
Updated
* Rebased with master
* Type inference
* private isCollapsed
* Removed default close
* New line in all files at end

**Comment by icfantv** [comment]
LGTM

**Pull request closed without merging**